api/src/main/java/uk/gov/dhsc/htbhf/claimant/entitlement/CycleEntitlementCalculator.java

```diff
@@ -43,15 +57,53 @@ private void validateArguments(Integer paymentCycleDurationInDays, Integer numbe
         }
     }
 
-    public PaymentCycleVoucherEntitlement calculateCycleVoucherEntitlement(Claimant claimant, List<LocalDate> dateOfBirthOfChildren) {
+    /**
+     * Calculates the total voucher entitlement for a payment cycle when there is no previous voucher entitlement.
+     * @param expectedDueDate expected due date
+     * @param dateOfBirthOfChildren the date of birth of the claimant's children
+     * @return the payment cycle voucher entitlement calculated for the claimant
+     */
+    public PaymentCycleVoucherEntitlement calculateEntitlement(Optional<LocalDate> expectedDueDate, List<LocalDate> dateOfBirthOfChildren) {
         List<LocalDate> entitlementDates = getEntitlementDates();
         List<VoucherEntitlement> voucherEntitlements = entitlementDates.stream()
-                .map(date -> entitlementCalculator.calculateVoucherEntitlement(claimant, dateOfBirthOfChildren, date))
+                .map(date -> entitlementCalculator.calculateVoucherEntitlement(expectedDueDate, dateOfBirthOfChildren, date))
                 .collect(Collectors.toList());
 
         return new PaymentCycleVoucherEntitlement(voucherEntitlements);
     }
 
+    /**
+     * Calculates the total voucher entitlement for a payment cycle.
```

Comment:
How about `Calculates the total voucher entitlement for a payment cycle given the previous entitlement` so it's clear what the difference is from the other method.

Reply:
Updated